### DISCUSSION TITLE
Use w/ Template on NonStandard Port

### DISCUSSION THREAD

**Discussion by alexbeau10** [selivan/https-ssl-cert-check-zabbix#19] [Q&A]
Hello,

Has anyone found a way to use this in a template via a macro to check SSL certs that are on a non-standard port?

It works great if I use the {HOST.DNS} macro in the item key but once I get one with a non-standard port I am at a loss as how to define the non-standard port and have it picked up by the item check.

Any help is appreciated. Thank you.

**Comment by selivan** [comment]
@alexbeau10 If you are using `userparameters_ssl_cert_check.conf` from the repository, then port is a second parameter, like:

```
ssl_cert_check_valid[{HOST.DNS},8443]
```

**Reply by alexbeau10** [reply]
@selivan, thank you for this great script and extension to Zabbix. I am using it as shown above, and works great. Maybe my work flow is not correct, but I created a template that includes an item that looks like this: ssl_cert_check_valid[{HOST.DNS}]. I have added this to a template that also includes a few triggers based on time till expiration. When I add a new host and apply this template it does not allow me to edit the item that is applied from the item. Do you know of a way to be able to edit the item if I need to add the port, or another place I can place the port in the host config and pull it with a macro? Does that make sense?

**Reply by selivan** [reply]
@alexbeau10

I think you can use macroses. Define `{$TLS_CHECK_PORT}=443` macros in template, use it:

`ssl_cert_check_valid[{HOST.DNS},{$TLS_CHECK_PORT}]`

Then if you define the same macros on host level, it will override template level macros, if I get zabbix documentation correctly:

https://www.zabbix.com/documentation/5.0/manual/config/macros/user_macros

**Reply by alexbeau10** [reply]
@selivan Thank you! This is exactly what I needed and works great. I was just unsure of the terms and couldn't find what I needed in the Zabbix docs. Thank you for pointing me in the right direction.
